src/messageschannel.cpp

```diff
@@ -152,26 +152,33 @@ void MatrixMessagesChannel::processMessageEvent(const QMatrixClient::RoomMessage
     qDebug().noquote() << Q_FUNC_INFO << "Process message" << doc.toJson(QJsonDocument::Indented);
     Tp::MessagePart header;
     header[QStringLiteral("message-token")] = QDBusVariant(event->id());
-    header[QStringLiteral("message-sent")]  = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
-    header[QStringLiteral("message-received")] = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
     if (event->senderId() == m_connection->matrix()->user()->id()) {
-        header[QStringLiteral("message-sender")]   = QDBusVariant(m_connection->selfHandle());
+        header[QStringLiteral("message-sender")]    = QDBusVariant(m_connection->selfHandle());
         header[QStringLiteral("message-sender-id")] = QDBusVariant(m_connection->selfID());
     } else {
-        header[QStringLiteral("message-sender")]   = QDBusVariant(m_connection->ensureContactHandle(event->senderId()));
+        header[QStringLiteral("message-sender")]    = QDBusVariant(m_connection->ensureContactHandle(event->senderId()));
         header[QStringLiteral("message-sender-id")] = QDBusVariant(event->senderId());
     }
+    header[QStringLiteral("message-sent")]  = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
+    header[QStringLiteral("message-received")] = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
+    if (event->isRedacted()) {
+        header[QStringLiteral("message-type")]    = QDBusVariant(Tp::ChannelTextMessageTypeDeliveryReport);
+        header[QStringLiteral("delivery-status")] = QDBusVariant(Tp::DeliveryStatusDeleted);
+        header[QStringLiteral("delivery-token")]  = QDBusVariant(event->transactionId());
```

Comment:
Delivery token is [an identifier for the message to which this delivery report refers](https://telepathy.freedesktop.org/spec/Channel_Interface_Messages.html#Simple-Type:Delivery_Report_Header_Key). It is valid only for a delivery report. In this case the message (should be) a normal message with known delivery status. It does not refer to anything. And the last issue is that transactionId() returns an empty string, but the delivery-token must not be empty.

Reply:
thank you! propably I have same issue with `MatrixMessagesChannel::onPendingEventChanged`

Reply:
changed from `transactionId` to `id` at `MatrixMessagesChannel::onPendingEventChanged`